### PR TITLE
Features/view spec n+1 problem

### PR DESCRIPTION
**Pull request by oguntayomicheal**
#### In this  PR, the following tasks were implemented:
- Add spec for all views
- Solve N+1 problem in post controller through eager loading

Before eager loading
Screenshot (224)

After eager loading
Screenshot (226)

### CODE REVIEW

File: spec/views/users_index_spec.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+      bio: 'john is a 21 year old footballer from Ghana',
+      posts_counter: 5
+    )
+  end
+
+  it 'displays users information' do
+    visit users_path
+
+    expect(page).to have_content(user1.name)
+    expect(page).to have_content(user2.name)
```
Comment (omar-muhamad) [inline spec/views/users_index_spec.rb]: - [ ] This test is failing while I run tests, Please make sure that all tests are passing.

image

File: spec/views/post_index_spec.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    expect(page).to have_content("Comments: #{post1.comments_counter}")
+    expect(page).to have_content("Comments: #{post2.comments_counter}")
+    expect(page).to have_content("Likes: #{post1.likes_counter}")
+    expect(page).to have_content("Likes: #{post2.likes_counter}")
+    expect(page).to have_content("Likes: #{post3.likes_counter}")
+
+    # When I click on a post, it redirects me to that post's show page.
+    click_link post1.title
+    expect(current_path).to eq(user_post_path(user, post1))
+  end
```
Comment (omar-muhamad) [inline spec/views/post_index_spec.rb]: - [ ] As per this [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/intergration_spec_and_n_1_project.md#:~:text=a%20post%20has.-,I%20can%20see%20a%20section%20for%20pagination%20if%20there%20are%20more%20posts%20than%20fit%20on%20the%20view.,-When%20I%20click), Please add a test that asserts you can see a section for pagination if there are more posts than fit on the view.

**Review by omar-muhamad — CHANGES_REQUESTED** [review]
# Changes Required ♻️
### Hi @oguntayomicheal and @MarwanDev,

### Nice job, you are almost there.

cat-driving-serious

## (Highlights) Good Points: :+1:
- Your work is well documented. ✔
- The correct workflow is used.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- [ ] Please check the inline comments below ⬇️.

## [Optional] suggestions:

- Nothing to mention.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: spec/views/users_index_spec.rb [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
```
Comment (christianbiring1) [inline spec/views/users_index_spec.rb:25]: - [ ] As suggested in the previous review, Kindly make sure you fix this error you are having in your test case. You may consider looking at the correct matcher to use here.
image

**Review by christianbiring1 — CHANGES_REQUESTED** [review]
Hi @oguntayomicheal , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.
# STATUS: REQUEST CHANGES :recycle: 

 ## Highlights
- [ ] Great work on implementing some of the changes as suggested by the previous reviewer :+1: 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by oguntayomicheal** [comment]
### Screenshot for the test
Hi @code reviewers, this is a screenshot showing the test are passing.

Screenshot (227)

**Review by rotimiazeez — APPROVED** [review]
Hi @oguntayomicheal 👏🏽, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- Good job implementing the suggested changes 👌🏽
- All test cases passed ✅️
- No linter error 👍🏼
- Good PR description and summary 👏🏾

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code. Some of them were simply missed by the previous reviewer, and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me(@rotimiazeez) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561), you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**